huxley/www/js/components/AdvisorFeedbackView.js

```diff
@@ -0,0 +1,195 @@
+              <thead>
+                <tr>
+                  <th>Committee</th>
+                  <th>Delegates</th>
+                  <th>1</th>
+                  <th>2</th>
+                  <th>3</th>
+                  <th>4</th>
+                  <th>Summary</th>
+                </tr>
+              </thead>
+              <tbody>
+                {this.renderAssignmentRows()}
+              </tbody>
+            </table>
+          </div>
+      </InnerView>
+    );
+  },
+
+  renderAssignmentRows: function() {
+    var committees = this.state.committees;
+    var countries = this.state.countries;
+    var assigned = this.state.assigned;
+    return this.state.assignments.map(function(assignment) {
+      var delegates = assigned[assignment.id];
+      if (delegates == null) {
+        return;
+      }
+      return (
+        <tr>
+          <td>{committees[assignment.committee].name}</td>
+          <td>{delegates[1] == 0 ?
+            delegates[0].name :
+            delegates[0].name + ', ' + delegates[1].name
+          }</td>
+          <td>
+            <input
+              className="choice"
+              type="checkbox"
+              checked={delegates[0].session_one}
+              disabled
+            />
+          </td>
+          <td>
+            <input
+              className="choice"
+              type="checkbox"
+              checked={delegates[0].session_two}
+              disabled
+            />
+          </td>
+          <td>
+            <input
+              className="choice"
+              type="checkbox"
+              checked={delegates[0].session_three}
+              disabled
+            />
+          </td>
+          <td>
+            <input
+              className="choice"
+              type="checkbox"
+              checked={delegates[0].session_four}
+              disabled
+            />
+          </td>
+          <textarea
+              className="text-input"
+              style={{"width": "95%"}}
+              defaultValue={delegates[0].published_summary}
+              disabled
+            />
+        </tr>
+      )
+    }.bind(this));
+  },
+
+
+  /*
+    To make it easier to assign and unassign delegates to assignments we maintain
+    a state variable called "assigned". "assigned" is a dictionary whose key is
+    an assignment id, and value is an array representing slots for two delegates
+    to be assigned to it. This way we can easily manage the relationship from
+    assignment to delegates via this object.
+  */
+
+  prepareAssignedDelegates: function(delegates) {
+    var assigned = {};
+    for (var i = 0; i < delegates.length; i++) {
+      if (delegates[i].assignment) {
+        if (assigned[delegates[i].assignment]) {
+          assigned[delegates[i].assignment][1] = delegates[i]; 
+        } else {
+          var slots = [0, 0, 0];
+          slots[0] = delegates[i];
+          slots[2] = delegates[i].assignment;
```

Comment:
I don't think you guys end up using this third slot. Additionally, if we remove the need to label each row by names instead of country we can simply this a lot since the information is duplicated on each delegate. Essentially this boils down to 
```
assigned[delegates[i].assignment] = delegates[i];
```